base/ca/src/main/java/org/dogtagpki/server/ca/rest/base/ProfileBase.java

```diff
@@ -0,0 +1,1253 @@
+                auditParams.put("inputs", StringUtils.join(auditDelete, ","));
+                auditProfileChange(
+                        ScopeDef.SC_PROFILE_INPUT,
+                        OpDef.OP_DELETE,
+                        profile.getId(),
+                        ILogger.SUCCESS,
+                        auditParams);
+            }
+
+            if (!auditAdd.isEmpty()) {
+                Map<String, String> auditParams = new LinkedHashMap<>();
+                auditParams.put("inputs", StringUtils.join(auditAdd, ","));
+                auditProfileChange(
+                        ScopeDef.SC_PROFILE_INPUT,
+                        OpDef.OP_ADD,
+                        profile.getId(),
+                        ILogger.SUCCESS,
+                        auditParams);
+            }
+
+            if (!auditModify.isEmpty()) {
+                Map<String, String> auditParams = new LinkedHashMap<>();
+                auditParams.put("inputs", StringUtils.join(auditModify, ","));
+                auditProfileChange(
+                        ScopeDef.SC_PROFILE_INPUT,
+                        OpDef.OP_MODIFY,
+                        profile.getId(),
+                        ILogger.SUCCESS,
+                        auditParams);
+            }
+        } catch (EProfileException e) {
+            Map<String, String> auditParams = new LinkedHashMap<>();
+
+            auditParams.put("added", StringUtils.join(auditAdd, ","));
+            auditParams.put("deleted", StringUtils.join(auditDelete, ","));
+            auditParams.put("modified", StringUtils.join(auditModify, ","));
+            auditProfileChange(
+                    ScopeDef.SC_PROFILE_INPUT,
+                    OpDef.OP_MODIFY,
+                    profile.getId(),
+                    ILogger.FAILURE,
+                    auditParams);
+            throw e;
+        }
+    }
+
+    private boolean isProfileAccessLimited(Principal principal) {
+        AuthzSubsystem authzSubsystem = engine.getAuthzSubsystem();
+        if (principal == null)
+            return true;
+        AuthToken authToken = null;
+        String authzMgrName = null;
+        if (principal instanceof PKIPrincipal pkiPrincipal) {
+            authzMgrName = "DirAclAuthz";
+            authToken = pkiPrincipal.getAuthToken();
+        } else {
+            String realm = null;
+            String[] parts = principal.getName().split("@", 2);
+            if (parts.length == 2) {
+                realm = parts[1];
+            }
+            try {
+                authzMgrName = authzSubsystem.getAuthzManagerNameByRealm(realm);
+            } catch (EAuthzUnknownRealm e) {
+                logger.error("Cannot find AuthzManager for external principal {}", principal.getName());
+                return true;
+            }
+            authToken = new ExternalAuthToken((GenericPrincipal) principal);
+        }
+        if (authToken == null)
+            return true;
```

Comment:
Could we move this into the `if` clause (between line 1197 and 1198) since in the `else` clause it cannot be `null`?